### PR TITLE
LPD-26164

### PR DESCRIPTION
**Pull request by icgsbr**
https://liferay.atlassian.net/browse/LPD-26164

### CODE REVIEW

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by icgsbr** [comment]
ci:test:sf

**Comment by icgsbr** [comment]
ci:test:sf

File: modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java [hunk 1]
```diff
@@ -704,6 +704,13 @@ private Object _getValue(
 				_log.debug(portalException);
 			}
 
+			if (Objects.equals(entityType, EntityField.Type.ID) &&
+				Objects.equals(DBManagerUtil.getDBType(), DBType.POSTGRESQL) &&
+				Validator.isNumber(String.valueOf(right))) {
+
+				return GetterUtil.getLong(right);
+			}
```
Comment (carolmariaabb) [inline modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java:712]: @icgsbr I won't have time to review the test code for the next three days, so please **don't forward** this PR yet. However, it is okay to create a hotfix with this change. Thanks!

**Comment by carolmariaabb** [comment]
@icgsbr please see https://github.com/liferay-objects/liferay-portal/pull/3381/

**Pull request closed without merging**